pkg/drivers/kic/kic.go

```diff
@@ -0,0 +1,250 @@
+		Role:              "control-plane",
+		ExtraMounts:       []oci.Mount{},
+		ExtraPortMappings: []oci.PortMapping{},
+		APIServerAddress:  "127.0.0.1", // medyagh:TODO make configurable
+		APIServerPort:     d.APIServerPort,
+		IPv6:              false, // MEDYA:TODO add proxy envs here
+	}
+
+	_, err := ks.Create(command.NewKICRunner(d.MachineName, d.OciBinary))
+	if err != nil {
+		return errors.Wrap(err, "create kic from spec")
+	}
+	return nil
+}
+
+// DriverName returns the name of the driver
+func (d *Driver) DriverName() string {
+	if d.OciBinary == "podman" {
+		return "podman"
+	}
+	return "docker"
+}
+
+// GetIP returns an IP or hostname that this host is available at
+func (d *Driver) GetIP() (string, error) {
+	node, err := node.Find(d.MachineName, d.exec)
+	if err != nil {
+		return "", fmt.Errorf("ip not found for nil node")
+	}
+	ip, _, err := node.IP()
+	return ip, err
+}
+
+// GetSSHHostname returns hostname for use with ssh
+func (d *Driver) GetSSHHostname() (string, error) {
+	return "", fmt.Errorf("driver does not have SSHHostName")
+}
+
+// GetSSHPort returns port for use with ssh
+func (d *Driver) GetSSHPort() (int, error) {
+	return 0, fmt.Errorf("driver does not support GetSSHPort")
+}
+
+// GetURL returns ip of the container running kic control-panel
+func (d *Driver) GetURL() (string, error) {
+	return d.GetIP()
+}
+
+// GetState returns the state that the host is in (running, stopped, etc)
+func (d *Driver) GetState() (state.State, error) {
+	cmd := exec.Command(d.OciBinary, "inspect", "-f", "{{.State.Status}}", d.MachineName)
+	out, err := cmd.CombinedOutput()
+	o := strings.Trim(string(out), "\n")
+	if err != nil {
+		return state.Error, errors.Wrapf(err, "error stop node %s", d.MachineName)
+	}
+	if o == "running" {
+		return state.Running, nil
+	}
+	if o == "exited" {
+		return state.Stopped, nil
+	}
+	if o == "paused" {
+		return state.Paused, nil
+	}
+	if o == "restarting" {
+		return state.Starting, nil
+	}
+	if o == "dead" {
+		return state.Error, nil
+	}
+	return state.None, fmt.Errorf("unknown state")
+
+}
+
+// Kill stops a host forcefully, including any containers that we are managing.
+func (d *Driver) Kill() error {
+	cmd := exec.Command(d.OciBinary, "kill", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "killing kic node %s", d.MachineName)
+	}
+	return nil
+}
+
+// Remove will delete the Kic Node Container
+func (d *Driver) Remove() error {
+	if _, err := d.nodeID(d.MachineName); err != nil {
+		return errors.Wrapf(err, "not found node %s", d.MachineName)
+	}
+	cmd := exec.Command(d.OciBinary, "rm", "-f", "-v", d.MachineName)
+	if err := cmd.Run(); err != nil {
+		return errors.Wrapf(err, "error removing node %s", d.MachineName)
+	}
+	return nil
+}
+
+// Restart a host
+func (d *Driver) Restart() error {
+	s, err := d.GetState()
+	if err != nil {
+		return errors.Wrap(err, "get kic state")
+	}
+	if s == state.Paused {
+		return d.Unpause()
```

Comment:
Should this restart the host too, similar to if the node is in state.Running? It seems like restart and unpause are different.

Reply:
I agree once we have a pause/resume command for minikube, would make restart a real restart. and pause/resume use this mechanisim.